### DISCUSSION TITLE
Ane Swift on iOS : IPA size is too Big

### DISCUSSION THREAD

**Discussion by pol2095** [airsdk/Adobe-Runtime-Support#2973] [Q&A]
Hello,

I created an *.ipa including an ANE using Swift on iOS, the same ANE using 
- only Objective-C : 5000 Ko
- using Swift : 68.000 Ko, for working I need to include these *.dylib to the *.ipa
![dylib](https://github.com/airsdk/Adobe-Runtime-Support/assets/6962232/e7adeffc-2aae-469f-89c7-2ce21f7bc735)

is-it possible to reduce it ?
Thanks

**Comment by ajwfrost** [comment]
Do you mean, to reduce the size of the ANE file? I would have thought you don't need to manually include those libraries into the ANE itself, they're provided as part of the AIR SDK so you could just add the linker commands for them..?

**Reply by pol2095** [reply]
I tried in the last versions since Swift 5 it's not necessary to add Swift dylib and the size is equals to obj-c simple library.

**Reply by marchbold** [reply]
dylibs aren't required in the ANE, however if you are supporting older iOS versions (older than around v12.4 I think) you need to include them in the Frameworks folder of your application. Generally we just suggest people set a minimum version of 12.4+ to avoid this.

**Reply by pol2095** [reply]
@Marchbold
Thanks for that clarification.

**Reply by pol2095** [reply]
I don't understand why it's necessary to add the Swift *.framework library in the ANE (packagedDependency) and in the "Frameworks" directory of the IPA ?

**Reply by marchbold** [reply]
Generally swift frameworks are dynamic which are required to be packaged in the `Frameworks` folder of your application. AIR includes this special folder in your application and signs in appropriately.
